src/boards/include/boards/waveshare_rp2350_usb_a.h

```
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+
+#ifndef _BOARDS_WAVESHARE_RP2350_USB_A_H
+#define _BOARDS_WAVESHARE_RP2350_USB_A_H
+
+pico_board_cmake_set(PICO_PLATFORM, rp2350)
+
+// For board detection
+#define WAVESHARE_RP2350_USB_A
+
+// On some samples, the xosc can take longer to stabilize than is usual
+#ifndef PICO_XOSC_STARTUP_DELAY_MULTIPLIER
+#define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 64
+#endif
+
+// --- RP2350 VARIANT ---
+#define PICO_RP2350A 1
+
+// --- UART ---
+#ifndef PICO_DEFAULT_UART
+#define PICO_DEFAULT_UART 0
+#endif
+#ifndef PICO_DEFAULT_UART_TX_PIN
+#define PICO_DEFAULT_UART_TX_PIN 0
+#endif
+#ifndef PICO_DEFAULT_UART_RX_PIN
+#define PICO_DEFAULT_UART_RX_PIN 1
+#endif
+
+// --- WS2812 ---
+#ifndef PICO_DEFAULT_WS2812_PIN
+#define PICO_DEFAULT_WS2812_PIN 16
+#endif
+
+// --- I2C ---
+#ifndef PICO_DEFAULT_I2C
+#define PICO_DEFAULT_I2C 1
+#endif
+#ifndef PICO_DEFAULT_I2C_SDA_PIN
+#define PICO_DEFAULT_I2C_SDA_PIN 6
+#endif
+#ifndef PICO_DEFAULT_I2C_SCL_PIN
+#define PICO_DEFAULT_I2C_SCL_PIN 7
+#endif
+
+// --- SPI ---
+#ifndef PICO_DEFAULT_SPI
+#define PICO_DEFAULT_SPI 0
+#endif
+#ifndef PICO_DEFAULT_SPI_SCK_PIN
+#define PICO_DEFAULT_SPI_SCK_PIN 2
+#endif
+#ifndef PICO_DEFAULT_SPI_TX_PIN
+#define PICO_DEFAULT_SPI_TX_PIN 3
+#endif
+#ifndef PICO_DEFAULT_SPI_RX_PIN
+#define PICO_DEFAULT_SPI_RX_PIN 4
+#endif
+#ifndef PICO_DEFAULT_SPI_CSN_PIN
+#define PICO_DEFAULT_SPI_CSN_PIN 5
+#endif
+
+// --- PIO USB ---
+#ifndef PICO_DEFAULT_PIO_USB_DP_PIN
+#define PICO_DEFAULT_PIO_USB_DP_PIN 12
```

There's no SDK support for this, so IMHO prefixing it with `PICO_DEFAULT_` is a bit misleading. Perhaps `WAVESHARE_RP2350_USB_A_PIO_USB_DP_PIN` would be better?
Also, if you're defining the `PIO_USB_DP_PIN`, would it make sense to also define the `PIO_USB_DM_PIN` ??

---

Is this define used by Pico-PIO-USB, if so im fine with it

---

ok, i guess it doesn't ... i'd go with `PIO_USB_DP_PIN` and then maybe encourage Pico-PIO-USB to provide a confif that uses those along with 

```
#ifndef PIO_USB_DP_PIN
#define PIO_USB_DP_PIN PIO_USB_DP_PIN_DEFAULT
#endif

---

Doesn't look like it? https://github.com/search?q=repo%3Asekigon-gonnoc%2FPico-PIO-USB%20PICO_DEFAULT_PIO_USB_DP_PIN&type=code

---

That macro is the one used in TinyUSB for the Pico PIO implementation.

I agree it may be a bit misleading. For further context, as far as I know, the only other RP2040 board with a USB-A port is the Adafruit RP2040 USB Host, and it doesn’t define anything specific for the USB-A port in its board header.

TinyUSB sets the default pins for that board [here](https://github.com/hathach/tinyusb/blob/master/hw/bsp/rp2040/boards/adafruit_feather_rp2040_usb_host/board.h).

It might be worth adding the custom board header to the TinyUSB repo instead?

---

Any chance we could move that bit into a seperate PR and merge the rest?

---

sure, go ahead